components/table/table-col-sort-button.js

```diff
@@ -108,20 +119,57 @@ export class TableColSortButton extends LocalizeCoreElement(FocusMixin(LitElemen
 		const buttonTitle = this.nosort
 			? undefined
 			: this.localize('components.table-col-sort-button.title', {
-				sourceType: this.sourceType,
-				direction: this.desc ? 'desc' : undefined
+				sourceType: this._hasDropdownItems && this._selectedMenuItemText ? 'value' : this.sourceType,
+				direction: this.desc ? 'desc' : undefined,
+				selectedMenuItemText: this._selectedMenuItemText
 			});
 
 		const iconView = !this.nosort ?
 			html`<d2l-icon icon="${this.desc ? 'tier1:arrow-toggle-down' : 'tier1:arrow-toggle-up'}"></d2l-icon>` :
 			null;
 
-		return html`<button
+		const button = html`<button
 				aria-describedby="${this._describedById}"
+				class="${classMap({ 'd2l-dropdown-opener': this._hasDropdownItems })}"
 				title="${ifDefined(buttonTitle)}"
 				type="button">
 				<slot></slot>${iconView}
 			</button><span id="${this._describedById}" hidden>${buttonDescription}</span>`;
+		if (this._hasDropdownItems) {
+			return html`<d2l-dropdown>
+					${button}
+					<d2l-dropdown-menu no-pointer align="start" vertical-offset="0">
+						<d2l-menu @d2l-menu-item-change="${this._handleMenuItemChange}">
+							<slot name="items" @slotchange="${this._handleSlotChange}"></slot>
+						</d2l-menu>
+					</d2l-dropdown-menu>
+				</d2l-dropdown>`;
+		} else {
+			return html`${button}<slot name="items" @slotchange="${this._handleSlotChange}"></slot>`;
+		}
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		// de-select any selected dropdown menu item
+		if (changedProperties.has('nosort') && this.nosort && this._hasDropdownItems) {
+			const selectedItem = this.querySelector('[selected]');
+			if (selectedItem) selectedItem.selected = false;
+		}
+	}
+
+	_handleMenuItemChange(e) {
+		this._selectedMenuItemText = e.target?.text;
+	}
+
+	_handleSlotChange(e) {
+		const items = e.target?.assignedNodes({ flatten: true }).filter((node) => node.nodeType === Node.ELEMENT_NODE);
+		const filteredItems = items.filter((item) => {
+			const role = item.getAttribute('role');
+			return (role === 'menuitem' || role === 'menuitemcheckbox' || role === 'menuitemradio');
```

Comment:
We do only want this to get `menu-item-radio` so perhaps we should only return those

Reply:
I was just going to ask about that. With the current slot API, where the consumer renders `d2l-menu-item-radio` elements, it almost invites them to try and put other types of menu-items into the slot. And then I was wondering, is that valid, do we want/need to prevent that?

Reply:
We only want `menu-item-radio` since we'd only want one multi-faceted sort option selected at a time.

Reply:
If we really want to lock it down, sometimes we do something like:
```
::slotted(*) {
	display: none;
}
::slotted(d2l-table-col-sort-button-item) {
	display: ...;
}
```
Technically I think they'd still be assigned in the slot and we'd want to still filter here, but developers would figure out pretty quick there stuff's not going to work.